### PR TITLE
Modify workflow to show deprecation progress against integration tests

### PR DESCRIPTION
**Pull request by GumpacG**
### Description
Legacy is disabled to force all integration tests to be run on the new engine. The test summary will be shown in GitHub Actions and can also be downloaded from the artifacts.
Screenshot 2023-01-17 at 3 25 37 PM

 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/1144
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java [hunk 1]
```diff
@@ -155,10 +155,9 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                             if (newSqlRequest.isExplainRequest()) {
                                 LOG.info("Request is falling back to old SQL engine due to: " + exception.getMessage());
                             }
-                            LOG.debug("[{}] Request {} is not supported and falling back to old SQL engine",
```
Comment (GabeFernandez310) [inline legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java:158]: Why was this changed?

Comment (GumpacG) [reply legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java:158]: Removed this because we are forcing it to throw an exception instead of falling back to legacy.

File: .github/workflows/sql-test-and-build-workflow.yml [hunk 2]
```diff
@@ -24,68 +24,44 @@ jobs:
...
-          - { os: macos-latest, java: 11, os_build_args: -x doctest -x integTest -x jacocoTestReport  }
-          - { os: ubuntu-latest, java: 17 }
-          - { os: windows-latest, java: 17, os_build_args: -x doctest -x integTest -x jacocoTestReport  -PbuildPlatform=windows }
-          - { os: macos-latest, java: 17, os_build_args: -x doctest -x integTest -x jacocoTestReport  }
+          - { os: ubuntu-latest, java: 11, os_build_args: -x doctest -x jacocoTestReport  }
     runs-on: ${{ matrix.entry.os }}
 
     steps:
-    - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
```
Comment (forestmvey) [inline .github/workflows/sql-test-and-build-workflow.yml:31]: Why the whitespace changes?

**Review by acarbonetto — APPROVED** [review]

**Review by forestmvey — APPROVED** [review]

File: .github/workflows/sql-test-and-build-workflow.yml (hunk 2 above)
Comment (GumpacG) [reply .github/workflows/sql-test-and-build-workflow.yml:31]: Unintentional. Will remove. Thanks!

**Review by MaxKsyunz — APPROVED** [review]
